lib/task_listener.js

```diff
@@ -253,6 +266,7 @@ export default class TaskListener extends EventEmitter {
       this.cleanupRunningState(runningState);
       return;
     }
+    await this.recordCapacity();
```

Comment:
Please add a comment that it's important to call `recordCapacity` before we remove the entry from `this.runningTasks` because it counts it... Just a single line comment will do, so we don't move it by accident.